bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/util/EventLogReader.java

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright (c) 2010-2025 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.satel.internal.util;
+
+import java.time.LocalDateTime;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.satel.internal.command.ReadDeviceInfoCommand.DeviceType;
+import org.openhab.binding.satel.internal.command.ReadEventCommand;
+import org.openhab.binding.satel.internal.command.ReadEventDescCommand;
+import org.openhab.binding.satel.internal.handler.SatelBridgeHandler;
+import org.openhab.binding.satel.internal.types.IntegraType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is a helper class for reading event log records.
+ *
+ * @author Krzysztof Goworek - Initial contribution
+ */
+@NonNullByDefault
+public class EventLogReader {
+
+    private static final String DETAILS_SEPARATOR = ", ";
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private final Map<String, @Nullable EventDescriptionCacheEntry> eventDescriptions = new ConcurrentHashMap<>();
+    private final SatelBridgeHandler bridgeHandler;
+    private final DeviceNameResolver deviceNameResolver;
+
+    public EventLogReader(SatelBridgeHandler bridgeHandler, DeviceNameResolver deviceNameResolver) {
+        this.bridgeHandler = bridgeHandler;
+        this.deviceNameResolver = deviceNameResolver;
+    }
+
+    /**
+     * Reads one record from the event log.
+     *
+     * @param eventIndex record index
+     * @return record description wrapped in {@linkplain Optional} or {@linkplain Optional#empty()} if there is no
+     *         record under given index or read command failed
+     */
+    public Optional<EventDescription> readEvent(int eventIndex) {
+        ReadEventCommand readEventCmd = new ReadEventCommand(eventIndex);
+        if (!bridgeHandler.sendCommand(readEventCmd, false)) {
+            logger.info("Unable to read event record for given index: {}", eventIndex);
+            return Optional.empty();
+        } else if (readEventCmd.isEmpty()) {
+            logger.info("No record under given index: {}", eventIndex);
```

Comment:
warn would be better. Please check others too. 
For reference: https://www.openhab.org/docs/developer/guidelines.html#f-logging